src/WebJobs.Script.WebHost/Diagnostics/DeferredLoggerProvider.cs

```diff
@@ -19,25 +21,24 @@ public class DeferredLogDispatcher : IDeferredLogDispatcher, IDisposable
             SingleWriter = false
         });
 
-        private readonly List<ILoggerProvider> _forwardingProviders = new(1);
         private bool _isEnabled = true;
         private bool _disposed = false;
 
         public int Count => _channel.Reader.Count;
 
-        bool IDeferredLogDispatcher.IsEnabled => _isEnabled;
-
-        public void AddLoggerProvider(ILoggerProvider provider)
-        {
-            _forwardingProviders.Add(provider);
-        }
-
-        public void Log(DeferredLogEntry log)
+        public ILogger CreateLogger(string categoryName)
         {
-            _channel.Writer.TryWrite(log);
+            if (_isEnabled)
+            {
+                return new DeferredLogger(_channel, categoryName);
+            }
+            else
+            {
+                return NullLogger.Instance;
+            }
         }
 
-        public void ProcessBufferedLogs(bool runImmediately = false)
+        public void ProcessBufferedLogs(IEnumerable<ILoggerProvider> forwardingProviders, bool runImmediately = false)
```

Comment:
nit: null check `forwardingProviders` - or coalesce to `Enumerable.Empty<>`

Reply:
Done.